docs: add link to RH docu VictoriaMetrics

schwesig opened the pull request:
[3.2. Exporting metrics to external endpoints](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.9/html/observability/using-observability#exporting-metrics-to-external-endpoints)

schwesig committed:
docs: add link to RH docu VictoriaMetrics

Signed-off-by: ​/Thor(sten)?/ Schwesig <89909507+schwesig@users.noreply.github.com>

larsks commented on architecture/observability/README.md:
```diff
@@ -57,6 +57,9 @@ The following technologies have been chosen for this solution:
 **Red Hat OpenShift Data Foundation (ODF):** ODF is an object storage that is required and made available for Thanos on the RHACM Hub cluster to store all the platform metrics collected from each of the managed clusters.
 
-**VictoriaMetrics:** VictoriaMetrics offers an option it is configured to go to a Backup Object Bucket, and on the infra side, the metrics go to a separate metrics bucket with retention.
+**VictoriaMetrics:** VictoriaMetrics offers an option it is configured to go to a Backup Object Bucket, and on the infra side, the metrics go to a separate metrics bucket with retention. [1]
```
Instead of using a footnote-style reference, just make the name a link:

```suggestion
**[VictoriaMetrics]:** VictoriaMetrics offers an option it is configured to go to a Backup Object Bucket, and on the infra side, the metrics go to a separate metrics bucket with retention.
```

larsks commented on architecture/observability/README.md:
```diff
@@ -57,6 +57,9 @@ The following technologies have been chosen for this solution:
 **Alertmanager:** The Alertmanager is a component of RHACM, and it takes care of deduplicating, grouping, and routing the alerts to the predefined appropriate end tools such as email, PagerDuty, or OpsGenie. Basically, the Alertmanager from the managed clusters will forward all the alerts to the RHACM hub cluster observability service for it to take appropriate action on the alerts.
+
+---
+[1]: https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.9/html/observability/using-observability#exporting-metrics-to-external-endpoints
```
```suggestion
[victoriametrics]: https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.9/html/observability/using-observability#exporting-metrics-to-external-endpoints
```

schwesig committed:
Update architecture/observability/README.md

Co-authored-by: Lars Kellogg-Stedman <lars@oddbit.com>

schwesig committed:
Update architecture/observability/README.md

Co-authored-by: Lars Kellogg-Stedman <lars@oddbit.com>

schwesig commented:
@larsks thanks; I first didn't think a small chapter would be fair to "block" the headline. But we don't have any other links yet, so this makes sense. I am using your idea.

larsks commented:
I think the link to https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.9/html/observability/using-observability#exporting-metrics-to-external-endpoints doesn't really explain what VictoriaMetrics **is**. Maybe the link should go to https://victoriametrics.com/products/open-source/ instead?

Also, I am having a hard time parsing the description:

> VictoriaMetrics offers an option it is configured to go to a Backup Object Bucket, and on the infra side, the metrics go to a separate metrics bucket with retention.

There are some grammar and clarity problems that need correcting there. I'm unable to offer a suggestion because I'm not entirely clear what this sentence is trying to say.

schwesig commented:
> I think the link to https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.9/html/observability/using-observability#exporting-metrics-to-external-endpoints doesn't really explain what VictoriaMetrics **is**. Maybe the link should go to https://victoriametrics.com/products/open-source/ instead?
> 
yes, I was thinking this, ... without opening a whole bucket and adding links to all mentioned services, apps, and operators.

schwesig commented:
Let me go through the text with Jeet and Harshil later and clarify a few things to make it easier to understand.

larsks committed:
Reformat observability documentation

Turn the list of technology choices into an actual list, and add links to
products.

larsks commented:
I've pushed up some suggested formatting changes. This doesn't address the language.

computate commented on architecture/observability/README.md:
```diff
@@ -43,20 +42,29 @@ The architecture diagram below is a graphical overview of the observability solu
+- **[Red Hat OpenShift Data Foundation][odf] (ODF)**: ODF is an object storage that is required and made available for Thanos on the RHACM Hub cluster to store all the platform metrics collected from each of the managed clusters.
 
-**Red Hat OpenShift Data Foundation (ODF):** ODF is an object storage that is required and made available for Thanos on the RHACM Hub cluster to store all the platform metrics collected from each of the managed clusters.
+- **[VictoriaMetrics]**: VictoriaMetrics offers an option it is configured to go to a Backup Object Bucket, and on the infra side, the metrics go to a separate metrics bucket with retention.
```
Instead of "VictoriaMetrics offers an option it is configured to go to a Backup Object Bucket", 
Maybe "VictoriaMetrics offers an option to also send metrics to a Backup Object Bucket"

schwesig replied:
accepted and changed

schwesig committed:
docs: updating VictoriaMetrics description

Review request computate; more clear explanation how VictoriaMetrics is
setup and can be used for our use case.

larsks commented on architecture/observability/README.md:
```diff
@@ -43,20 +42,29 @@ The architecture diagram below is a graphical overview of the observability solu
+- **[Red Hat OpenShift Data Foundation][odf] (ODF)**: ODF is an object storage that is required and made available for Thanos on the RHACM Hub cluster to store all the platform metrics collected from each of the managed clusters.
 
-**Red Hat OpenShift Data Foundation (ODF):** ODF is an object storage that is required and made available for Thanos on the RHACM Hub cluster to store all the platform metrics collected from each of the managed clusters.
+- **[VictoriaMetrics]**: VictoriaMetrics offers an option to also send metrics to a Backup Object Bucket, and on the infra side, the metrics go to a separate metrics bucket with retention.
```
This is still missing the answer to the question, "What is VictoriaMetrics?".  How about:

```suggestion
- **[VictoriaMetrics]**: VictoriaMetrics is an alternative storage backend for Prometheus. It offers an option to store metrics in an object storage service. On the infra side, the metrics go to a separate metrics bucket with a retention policy to limit the space consumed by metrics storage.
```

larsks commented on architecture/observability/README.md:
```diff
@@ -43,20 +42,29 @@ The architecture diagram below is a graphical overview of the observability solu
+- **[Grafana]**: Grafana is being utilized to provide dashboard(s) for visibility to the end-to-end platform metrics for all the clusters through a single pane of glass. Grafana comes out of the box as part of the RHACM Observability stack and is installed during RHACM installation process.
 
-**Grafana:** Grafana is being utilized to provide dashboard(s) for visibility to the end-to-end platform metrics for all the clusters through a single pane of glass. Grafana comes out of the box as part of the RHACM Observability stack and is installed during RHACM installation process.
+- **[Red Hat OpenShift Data Foundation][odf] (ODF)**: ODF is an object storage that is required and made available for Thanos on the RHACM Hub cluster to store all the platform metrics collected from each of the managed clusters.
```
Maybe:

```suggestion
- **[Red Hat OpenShift Data Foundation][odf] (ODF)**: ODF is a storage operator built on top of Ceph (for block storage) and Noobaa (for object storage). We use it to provide object storage for Thanos on the RHACM Hub cluster to store all the platform metrics collected from each of the managed clusters.
```

schwesig committed:
Update architecture/observability/README.md

Co-authored-by: Lars Kellogg-Stedman <lars@oddbit.com>

schwesig committed:
adding more details about VictoriaMetrics

schwesig commented:
@larsks thanks for all the links. I added some more benefits using VictoriaMetrics in the doc. I hope this answers the why a little bit better.
